EMS/core-bundle/src/Controller/User/GroupController.php

```diff
@@ -0,0 +1,152 @@
+        private readonly string $templateNamespace,
+    ) {
+    }
+
+    public function index(Request $request): Response
+    {
+        $table = $this->dataTableFactory->create(GroupDataTableType::class);
+
+        $form = $this->createForm(TableType::class, $table, [
+            'reorder_label' => t('type.reorder', ['type' => 'group'], 'emsco-core'),
+        ]);
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            match ($this->getClickedButtonName($form)) {
+                TableAbstract::DELETE_ACTION => $this->groupManager->deleteByIds($table->getSelected()),
+                default => $this->logger->messageError(t('log.error.invalid_table_action', [], 'emsco-core')),
+            };
+
+            return $this->redirectToRoute(Routes::GROUP_INDEX);
+        }
+
+        return $this->render("@$this->templateNamespace/crud/overview.html.twig", [
+            'form' => $form,
+            'title' => t('type.title_overview', ['type' => 'group'], 'emsco-core'),
+            'subTitle' => t('type.title_sub', ['type' => 'group'], 'emsco-core'),
+            'breadcrumb' => $this->breadcrumb(),
+        ]);
+    }
+
+    public function addGroup(Request $request): Response
+    {
+        $group = new Group();
+
+        $form = $this->createForm(GroupType::class, $group, ['mode' => UserType::MODE_CREATE]);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->groupManager->create($group);
+
+            return $this->redirectToRoute(Routes::GROUP_INDEX);
+        }
+
+        return $this->render("@$this->templateNamespace/group/create.html.twig", [
+            'form' => $form,
+            'title' => t('type.title_overview', ['type' => 'group'], 'emsco-core'),
+            'subTitle' => t('type.title_sub', ['type' => 'group'], 'emsco-core'),
+            'breadcrumb' => $this->breadcrumb(),
+        ]);
+    }
+
+    public function deleteGroup(Group $group): Response
+    {
+        $this->groupManager->deleteGroup($group);
+
+        return $this->redirectToRoute(Routes::GROUP_INDEX);
+    }
+
+    public function deleteAllGroup(): Response
+    {
+        $this->groupManager->deleteAllGroup();
+
+        return $this->redirectToRoute(Routes::GROUP_INDEX);
+    }
+
+    public function editGroup(Group $group, Request $request): Response
+    {
+        $userNotInGroupDataTable = $this->usersInGroupDataTable($request, $group, false);
+        $userGroupDataTable = $this->usersInGroupDataTable($request, $group, true);
+        $form = $this->createForm(GroupType::class, $group, [
+            'mode' => UserType::MODE_UPDATE,
+        ]);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->groupManager->editGroup($group);
+
+            return $this->redirectToRoute(Routes::GROUP_INDEX);
+        }
+
```

Comment:
```suggestion
        $form = $this->createForm(GroupType::class, $group, [
            'mode' => UserType::MODE_UPDATE,
        ]);
        $form->handleRequest($request);

        if ($form->isSubmitted() && $form->isValid()) {
            $this->groupManager->editGroup($group);

            return $this->redirectToRoute(Routes::GROUP_INDEX);
        }

        $userNotInGroupDataTable = $this->usersInGroupDataTable($request, $group, false);
        $userGroupDataTable = $this->usersInGroupDataTable($request, $group, true);
```
Building those datatable are useless in case of submit

Reply:
ok